library/exports.all

```diff
@@ -1,19 +1,25 @@
-(DEFINE-FILE-INFO PACKAGE "INTERLISP" READTABLE "XCL" BASE 10 FORMAT XCCS)
+<<<<<<< Updated upstream
+(DEFINE-FILE-INFO PACKAGE "INTERLISP" READTABLE "INTERLISP" BASE 10)
 (LISPXPRIN1 "EXPORTS GATHERED FROM {DSK}<Users>kaplan>Local>medley3.5>git-medley>sources> ON 
-30-Jul-2021 13:12:33"  T)
+19-Aug-2021 12:07:30"  T)
+=======
+(DEFINE-FILE-INFO PACKAGE "INTERLISP" READTABLE "INTERLISP" BASE 10 FORMAT XCCS)
+(LISPXPRIN1 "EXPORTS GATHERED FROM {DSK}<Users>kaplan>Local>medley3.5>git-medley>sources> ON 
+ 5-Aug-2021 21:04:18"  T)
+>>>>>>> Stashed changes
 (LISPXTERPRI T)
 (RPAQQ MODARITHMACROS (CEIL FLOOR FOLDHI FOLDLO MODUP UNFOLD MOD))
 (PUTPROPS CEIL MACRO ((X N) (FLOOR (IPLUS X (CONSTANT (SUB1 N))) N)))
 (PUTPROPS FLOOR MACRO ((X N) (LOGAND X (CONSTANT (LOGXOR (SUB1 N) -1)))))
 (PUTPROPS FOLDHI MACRO (X (PROG ((FORM (CAR X)) (DIVISOR (CAR (CONSTANTEXPRESSIONP (CADR X))))) (OR (
-AND DIVISOR (POWEROFTWOP DIVISOR)) (\\ILLEGAL.ARG (CADR X))) (RETURN (LIST (QUOTE LRSH) (LIST (QUOTE 
+AND DIVISOR (POWEROFTWOP DIVISOR)) (\ILLEGAL.ARG (CADR X))) (RETURN (LIST (QUOTE LRSH) (LIST (QUOTE 
```

Comment:
Not sure how this would change, unless the define-file-info changed.  But that doesn't show up as a diff.

We should probably look at how GATHEREXPORTS deals with reader environments, including the format.  With quick look, it seems like the call to \DEFINE-FILE-INFO should have TOFILE instead of NIL as its first argument, but that shouldn't affect the readtable.

Reply:
I've fixed GATHEREXPORTS so that it does install the the export.all's expected format in the TOFILE, no matter what the default format might have been for that filedevice.  But still I don't see how that would affect this.

 I'll also note that it searches for the begin-export-header using FFILEPOS, and that still needs to be fixed up so that it doesn't assume that it knows something about the external format (although the string in this case is ascii so it probably is always safe).  FFILEPOS and FILEPOS are separate issues.

Reply:
One other note:  It turns out that GATHEREXPORTS is called only by MAKE-EXPORTS.ALL in MEDLEY-UTILS.  This is driven by the variable EXPORTFILES in FILESETS.  So, if someone thinks that they are adding an export in a file that doesn't already have one, they have to know that FILESETS has to be updated.  That should go into some documentation somewhere, if it isn't already specified